src/common.ts

```diff
@@ -1,6 +1,83 @@
+import axios from 'axios';
+import cookie from 'cookie';
+import { User, updateProfile } from 'firebase/auth';
+import { doc, Firestore, setDoc } from 'firebase/firestore';
+import moment from 'moment';
+
+const TOKEN_COOKIE_ID = 'climate_iq_access_token';
+const EXPIRY_COOKIE_ID = 'climate_iq_access_token_expiry';
+const TOKEN_API_ENDPOINT_URI = 'https://34.49.55.140.nip.io/oauth/token';
+
 export const REDIRECT_URI_PROP = {
   redirectUri: {
     type: String,
     default: '',
   },
 };
+
+export function deleteCookies() {
+  document.cookie = cookie.serialize(TOKEN_COOKIE_ID, '', {
+    maxAge: -1,
+    path: '/',
+  });
+  document.cookie = cookie.serialize(EXPIRY_COOKIE_ID, '', {
+    maxAge: -1,
+    path: '/',
+  });
+}
+
+export async function getApigeeTokenAndSetCookies(
+  firebaseToken: string | null,
+) {
+  const cookies = cookie.parse(document.cookie);
+  if (cookies[TOKEN_COOKIE_ID]) {
+    if (parseInt(cookies[EXPIRY_COOKIE_ID]) > moment().unix()) {
+      return;
+    }
+  }
+  const response = await axios.post(
+    TOKEN_API_ENDPOINT_URI,
+    {
+      grant_type: 'urn:ietf:params:oauth:grant-type:jwt-bearer',
+      assertion: firebaseToken,
+    },
+    {
+      headers: {
+        Authorization:
+          'Basic cWNRZXNsbnY5eVFTS09kbkVEWDBCTkRVM2xjTzIwOWU6ZmFkVWZ5Q21OMDg0emJJcQ==',
```

Comment:
Here is the new authorization code to use: cDF1QXM4T3NGdDI2emlUYWIzQmt2cFdNOUJIbDVyUmNjT25CZHhzVFIya3MwVEZ0OmNTNU1WSFFjTkI4VHJ0UXJLOVZpc09BNDFNUGVRQUtjWHc5ZmRPcTZIWFN4aDBUdU1Va0h5Tk9wZ0ZsYTBpYnI=